lib/lintPostcssResult.mjs

```diff
@@ -95,7 +95,11 @@ export default async function lintPostcssResult(stylelintOptions, postcssResult,
 		// Log the rule's severity in the PostCSS result
 		const defaultSeverity = config.defaultSeverity || 'error';
 		// disableFix in secondary option
-		const disableFix = (secondaryOptions && secondaryOptions.disableFix === true) || false;
+		const disableFix = secondaryOptions?.disableFix || false;
+		const fileFix = !disableFix && config.fix && isFileFixCompatible;
+		// until #2643 is resolved
+		const ruleFix =
```

Comment:
medium

This condition should use `ruleMetadata` instead of `safeDisables`.
i.e. it's rule specific not file specific